crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1054,9 +1056,14 @@ impl<'db> TypeInferenceBuilder<'db> {
             .types
             .expression_ty(iterable.scoped_ast_id(self.db, self.scope));
 
-        let loop_var_value_ty = iterable_ty
-            .iterate(self.db)
-            .unwrap_with_diagnostic(iterable.into(), self);
+        let loop_var_value_ty = if is_async {
+            // TODO(Alex): async iterables/iterators!
```

Comment:
Heh. Dhruv just this morning told me via DM that we were encouraged to do this at Astral :) he linked [this](https://gist.github.com/dmnd/ed5d8ef8de2e4cfea174bd5dafcda382#whats-the-point-of-putting-your-name-on-a-todo) by way of explanation

I don't have a strong preference either way!

Reply:
Interesting. I see that rationale, and I am more OK with it if the interpretation is more "this person might have relevant context" and not "you shouldn't work on this unless you check with this person first." But I'm still concerned that the non-cookie-licking interpretation isn't necessarily clear to a new contributor reading the code.

Overall I still think I don't like it. If there's relevant context that's needed to understand the TODO, I'd rather that be in the comment with the TODO, rather than a person's name to go talk to. The case where you really need to go talk to a specific person to understand some code shouldn't be _that_ common, if we are writing reasonably understandable and well-commented code, and when that case arises, `git blame` exists. (That case can just as well arise for changes in an area of the code that doesn't have a TODO comment, so should we just comment every piece of code we write with our name, because `git blame` is too hard to use?)

Reply:
Opened a convo in Discord about this.

Reply:
@charliermarsh may have more context. He prefers adding the name. 

I agree that `git blame` may give you the same context. But it can be difficult if the code went through multiple refactors or was even moved to a different location.

Reply:
I don't really have a preference here, though I think it's pretty easy for code to move around enough to make `git blame` useless at our pace of refactoring and development. I wouldn't mind skipping the inclusion of a name.

Reply:
Before I was sent the above link, it was 100% my assumption that the meaning of the name in the TODO comment was "This is a TODO for me, i.e. I am going to fix it" (and thus, implicitly, nobody else should do it without asking me first.)

We can of course clarify this internally, but that doesn't mean it will be clear to the rest of the world.